From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <tobias.fischer@qut.edu.au>
Date: Fri, 30 Oct 2026 18:21:56 +1000
Subject: [PATCH] Add CCCORELIB_USE_SYSTEM_NANOFLANN to use an installed
 nanoflann

find_package(nanoflann) instead of the bundled extern/nanoflann submodule;
off by default.
---
 CMakeLists.txt | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a063738..1ef45b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,10 +116,15 @@ target_compile_definitions( CCCoreLib
 )
 
 # Nanoflann
-set( NANOFLANN_BUILD_BENCHMARKS OFF CACHE BOOL "" FORCE )
-set( NANOFLANN_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE )
-set( NANOFLANN_BUILD_TESTS OFF CACHE BOOL "" FORCE )
-add_subdirectory( extern/nanoflann EXCLUDE_FROM_ALL )
+option( CCCORELIB_USE_SYSTEM_NANOFLANN "Use an installed nanoflann instead of the bundled copy" OFF )
+if ( CCCORELIB_USE_SYSTEM_NANOFLANN )
+	find_package( nanoflann CONFIG REQUIRED )
+else()
+	set( NANOFLANN_BUILD_BENCHMARKS OFF CACHE BOOL "" FORCE )
+	set( NANOFLANN_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE )
+	set( NANOFLANN_BUILD_TESTS OFF CACHE BOOL "" FORCE )
+	add_subdirectory( extern/nanoflann EXCLUDE_FROM_ALL )
+endif()
 target_link_libraries( CCCoreLib
 	PUBLIC
 		nanoflann::nanoflann
-- 
2.53.0

